inc/zoo/swar/SWAR.h

```diff
@@ -123,15 +125,20 @@ template<typename T> constexpr typename std::make_unsigned<T>::type lsbIndex(T v
 /// SWAR operations are usually constant time, log(lane count) cost, or O(lane count) cost.
 /// Certain computational workloads can be materially sped up using SWAR techniques.
 template<int NBits, typename T = uint64_t> struct SWAR {
+    static constexpr inline auto LaneCount = sizeof(T) * 8 / NBits;
     SWAR() = default;
     constexpr explicit SWAR(T v): m_v(v) {}
     constexpr explicit operator T() const noexcept { return m_v; }
 
     constexpr T value() const noexcept { return m_v; }
 
-    constexpr SWAR operator|(SWAR o) const noexcept { return SWAR(m_v | o.m_v); }
-    constexpr SWAR operator&(SWAR o) const noexcept { return SWAR(m_v & o.m_v); }
-    constexpr SWAR operator^(SWAR o) const noexcept { return SWAR(m_v ^ o.m_v); }
+    constexpr SWAR operator~() const noexcept { return SWAR{~m_v}; }
+    #define SWAR_BINARY_OPERATORS_X_LIST \
+      X(SWAR, &) X(SWAR, ^) X(SWAR, |) X(SWAR, -) X(SWAR, +) X(SWAR, *)
+    #define X(rt,op) constexpr rt operator op (rt o) const noexcept { return rt (m_v op o.m_v); };
+    SWAR_BINARY_OPERATORS_X_LIST
+    #undef X
+    #undef SWAR_BINARY_OPERATORS_X_LIST
```

Comment:
You may let the X list be defined.  There may be uses afterward.
On the other hand, as the code does, once done with a particular `X` macro, it needs to be undefined